sonar-application/src/test/java/org/sonar/application/ClusterTest.java

```diff
@@ -0,0 +1,163 @@
+        ).isEqualTo(true);
+        assertThat(
+          cluster.hazelcastInstance.getConfig().getNetworkConfig().getInterfaces().getInterfaces()
+        ).containsExactly(ipAddress);
+        assertThat(
+          ((InetSocketAddress) cluster.hazelcastInstance.getLocalEndpoint().getSocketAddress()).getPort()
+        ).isEqualTo(port);
+        assertThat(
+          ((InetSocketAddress) cluster.hazelcastInstance.getLocalEndpoint().getSocketAddress()).getAddress().getHostAddress()
+        ).isEqualTo(ipAddress);
+      }
+    }
+  }
+
+  @Test
+  public void test_with_already_used_port() throws IOException {
+    ClusterProperties clusterProperties = initDefaultProperties();
+    InetAddress ipAddress = findIPv4Address();
+
+    Assume.assumeTrue(ipAddress != null);
+    ServerSocket socket = null;
+    Cluster cluster = null;
+    try {
+      socket = new ServerSocket(0, 100, ipAddress);
+      clusterProperties.setInterfaces(ipAddress.getHostAddress());
+      clusterProperties.setPort(socket.getLocalPort());
+      expectedException.expect(HazelcastException.class);
+      cluster = new Cluster(clusterProperties);
+    } finally {
+      if (socket != null) {
+        socket.close();
+      }
+      if (cluster != null) {
+        cluster.hazelcastInstance.getLifecycleService().terminate();
+      }
+    }
+  }
+
+  @Test
+  public void test_adding_port_to_members() throws Exception {
+    ClusterProperties clusterProperties = initDefaultProperties();
+    InetAddress ipAddress = findIPv4Address();
+
+    Assume.assumeTrue(ipAddress != null);
+    clusterProperties.setMembers(ipAddress.getHostAddress());
+
+
+    try (Cluster cluster = new Cluster(clusterProperties)) {
+      assertThat(
+        cluster.hazelcastInstance.getConfig().getNetworkConfig().getJoin().getTcpIpConfig().getMembers()
+      ).containsExactly(ipAddress.getHostAddress() + ":9003");
+    }
+  }
+
+  private ClusterProperties initDefaultProperties() {
+    Props props = new Props(new Properties());
+    props.set(ClusterProperties.Parameter.ENABLED.getName(), "true");
+    return new ClusterProperties(props);
+  }
+
+  private InetAddress findIPv4Address() throws SocketException {
+    Enumeration<NetworkInterface> interfaces = NetworkInterface.getNetworkInterfaces();
+
+    while (interfaces.hasMoreElements()) {
+      NetworkInterface networkInterface = interfaces.nextElement();
+      if (!networkInterface.isVirtual()
+        && !networkInterface.isLoopback()
+        && !networkInterface.getName().startsWith("docker")) {
+        Enumeration<InetAddress> ips = networkInterface.getInetAddresses();
+        while (ips.hasMoreElements()) {
+          InetAddress ip = ips.nextElement();
+
+          if (ip instanceof Inet4Address) {
+            return ip;
+          }
+        }
+      }
+    }
+
+    return null;
+  }
+
+  private int findAvailablePort() throws IOException {
+    try (ServerSocket ignored = new ServerSocket(0, 100)) {
```

Comment:
My bad, I was looking at Hazelcast code and it was their default. It's not useful.